### PR TITLE
docs: Polish the C++ docs

### PR DESCRIPTION
**Pull request by hunger**
Just some typo/vale/possible language fixes with a few corrections and updates thrown in for good measure.

I tried to keep things small and self-contained last time round, but that did not help with doc review, so I did not bother this time round.

### CODE REVIEW

**Review by ogoffart — APPROVED** [review]
Nice changes. Just some minor comments.


I wonder if we should document somewhere how to build in release mode https://github.com/slint-ui/slint-cpp-template/issues/8

File: api/cpp/docs/getting_started.md [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
+   the `.slint` files to C++ source code and included that into your
+   built target.
+2. The generated C++ source code needs the Slint run-time library. Use
+   `target_link_libraries` to link your build target to `Slint::Slint`.
 
-A typical example looks like this:
+A minimal CMake `CMakeLists.txt` file looks like this:
 
 ```cmake
 cmake_minimum_required(VERSION 3.19)
```
[the remark this answers is not in the document]
Comment (ogoffart) [reply api/cpp/docs/getting_started.md:18]: Actually I just checked and this was added recently in commit 3c65c6177d742b7a81d8e723c421f861b8392b0d

And that actually need cmake 3.21

```suggestion
cmake_minimum_required(VERSION 3.21)
```

File: docs/recipes/recipes.md [hunk 2]
```diff
@@ -1,10 +1,12 @@
+<!-- cSpell: ignore dgettext ngettext xgettext -->
 # Recipes and Examples
```
[the remark this answers is not in the document]
Comment (ogoffart) [reply docs/recipes/recipes.md:2]: Doesn't have to be.

File: docs/recipes/recipes.md [hunk 3]
```diff
@@ -1,10 +1,12 @@
...
 
-This page provides a collection of common use-cases and how to implement them using Slint.
+This page provides a collection of common use-cases and how to implement them
+using Slint.
 
 ## Get Started
 
-### A clickable Button
+### A Click-Able Button
```
[the remark this answers is not in the document]
Comment (tronical) [reply docs/recipes/recipes.md:9]: Please don't trust the tool blindly. It's clickable :)

File: api/cpp/docs/cmake.md [hunk 4]
```diff
@@ -37,80 +45,98 @@ First you need to install the prerequisites:
...
+You can include Slint into your CMake project using CMake's [`FetchContent`](https://cmake.org/cmake/help/latest/module/FetchContent.html) feature.
+Insert the following snippet into your `CMakeLists.txt` to make CMake download the latest 1.x release, compile it and make the CMake integration available:
 
 ```cmake
 include(FetchContent)
 FetchContent_Declare(
     Slint
     GIT_REPOSITORY https://github.com/slint-ui/slint.git
-    GIT_TAG release/0.3
+    GIT_TAG release/1
```
[the remark this answers is not in the document]
Comment (ogoffart) [reply api/cpp/docs/cmake.md]: We've used a branch the whole time and it worked well.  Where did you see that branch was not supported.  The docs say: (emphasis mine)  https://cmake.org/cmake/help/latest/module/ExternalProject.html#command:externalproject_add

> `GIT_TAG <tag>`
>   Git **branch** name, tag or commit hash [...]

**Review by ogoffart — COMMENTED** [review]
Please revert the part that change it to use the version tag, we should use a branch.

**Review by ogoffart — APPROVED** [review]

**Pull request merged**